### PR TITLE
Fix dupe leaf metrics

### PR DESCRIPTION
**Pull request by AlCutter**
Addresses #1057

### CODE REVIEW

File: server/log_rpc_server.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func (t *TrillianLogRPCServer) QueueLeaves(ctx context.Context, req *trillian.Qu
 	}
 
 	for i, existingLeaf := range ret {
-		if existingLeaf != nil {
+		if existingLeaf.Status.Code == codes.AlreadyExists {
```
Comment (daviddrysdale) [inline server/log_rpc_server.go]: need both old and new code together I think

Comment (daviddrysdale) [reply server/log_rpc_server.go]: (plus a cast, like wot Travis said)

**Comment by AlCutter** [comment]
Turned out to be a little more involved - PTAL?

File: server/log_rpc_server.go [hunk 2]
```diff
@@ -132,13 +132,11 @@ func (t *TrillianLogRPCServer) QueueLeaves(ctx context.Context, req *trillian.Qu
...
-		if existingLeaf != nil {
-			// There was a pre-existing leaf.
-			t.leafCounter.Inc("existing")
-		} else {
-			ret[i] = &trillian.QueuedLogLeaf{Leaf: req.Leaves[i], Status: status.Convert(nil).Proto()}
+	for _, l := range ret {
+		if l.Status == nil || l.Status.Code == int32(codes.OK) {
 			t.leafCounter.Inc("new")
+		} else if l.Status.Code == int32(codes.AlreadyExists) {
+			t.leafCounter.Inc("existing")
```
Comment (daviddrysdale) [inline server/log_rpc_server.go]: (Idle musing) If we had an `} else {` arm here, what would we do?

**Pull request merged**